packages/astro/src/content/types-generator.ts

```diff
@@ -515,8 +515,16 @@ async function writeContentFiles({
 		contentConfig ? `typeof import(${configPathRelativeToCacheDir})` : 'never',
 	);
 
-	await fs.promises.writeFile(
-		new URL(CONTENT_TYPES_FILE, settings.dotAstroDir),
-		typeTemplateContent,
-	);
+	if (settings.injectedTypes.some((t) => t.filename === CONTENT_TYPES_FILE)) {
```

Comment:
What does this do?

Reply:
If it's the first time, we inject types the usual way. `sync` will handle creating files and references. If it's not the first time, we just override the dts content. Does it make sense to you?

Reply:
I think a comment would be great on top just explaining that would be great! This file is already quite tough to read IMO

Reply:
```suggestion
	// If it's the first time, we inject types the usual way. sync() will handle creating files and references. If it's not the first time, we just override the dts content
	if (settings.injectedTypes.some((t) => t.filename === CONTENT_TYPES_FILE)) {
```